From 774c1e12c3d854825aa02147e6aa661b9caf4fe4 Mon Sep 17 00:00:00 2001
From: Robby Cai <r63905@freescale.com>
Date: Fri, 22 Aug 2008 14:29:50 +0800
Subject: [PATCH 043/449] ENGR00088673 update i.MX37 VPU firmware to v1.0.5,
 VPU package to v1.2.6

Accordingly, add vpu_DecClrDispFlag() to clear the busy flag of previous
dequeued frame buffer if VPU meets the case -- no picture to be displayed,
i.e., outputInfo.indexFrameDisplay == -3 or -2.

Signed-off-by: Robby Cai <r63905@freescale.com>
---
 test/mxc_vpu_test/dec.c      | 20 +++++++++++++-------
 test/mxc_vpu_test/utils.c    |  2 ++
 test/mxc_vpu_test/vpu_test.h |  1 -
 3 files changed, 15 insertions(+), 8 deletions(-)

diff --git a/test/mxc_vpu_test/dec.c b/test/mxc_vpu_test/dec.c
index 786cebc..a2f0c88 100644
--- a/test/mxc_vpu_test/dec.c
+++ b/test/mxc_vpu_test/dec.c
@@ -769,6 +769,13 @@ decoder_start(struct decode *dec)
 			warn_msg("VPU doesn't have picture to be displayed.\n"
 				"\toutinfo.indexFrameDisplay = %d\n",
 						outinfo.indexFrameDisplay);
+			if (disp_clr_index >= 0) {
+				err = vpu_DecClrDispFlag(handle, disp_clr_index);
+				if (err)
+					err_msg("vpu_DecClrDispFlag failed Error code"
+							" %d\n", err);
+			}
+			disp_clr_index = outinfo.indexFrameDisplay;
 			continue;
 		}
 
@@ -791,13 +798,13 @@ decoder_start(struct decode *dec)
 			if (err)
 				return -1;
 
-			if (disp_clr_index != -1) {
+			if (disp_clr_index >= 0) {
 				err = vpu_DecClrDispFlag(handle, disp_clr_index);
 				if (err)
 					err_msg("vpu_DecClrDispFlag failed Error code"
 							" %d\n", err);
-                        }
-                        disp_clr_index = disp->buf.index;
+			}
+			disp_clr_index = disp->buf.index;
 		} else {
 			pfb = pfbpool[outinfo.indexFrameDisplay];
 
@@ -819,7 +826,7 @@ decoder_start(struct decode *dec)
 				}
 			}
 
-			if (disp_clr_index != -1) {
+			if (disp_clr_index >= 0) {
 				err = vpu_DecClrDispFlag(handle,disp_clr_index);
 				if (err)
 					err_msg("vpu_DecClrDispFlag failed Error code"
@@ -1119,12 +1126,11 @@ decoder_parse(struct decode *dec)
 	 * used for decoder again. One framebuffer dequeue from IPU is delayed
 	 * for performance improvement and one framebuffer is delayed for
 	 * display flag clear.
-
+	 *
 	 * Performance is better when more buffers are used if IPU performance
 	 * is bottleneck.
 	 */
-
-	dec->fbcount = initinfo.minFrameBufferCount+2;
+	dec->fbcount = initinfo.minFrameBufferCount + 2;
 	dec->picwidth = ((initinfo.picWidth + 15) & ~15);
 	dec->picheight = ((initinfo.picHeight + 15) & ~15);
 	if ((dec->picwidth == 0) || (dec->picheight == 0))
diff --git a/test/mxc_vpu_test/utils.c b/test/mxc_vpu_test/utils.c
index fefa9da..57ad356 100644
--- a/test/mxc_vpu_test/utils.c
+++ b/test/mxc_vpu_test/utils.c
@@ -370,6 +370,7 @@ open_files(struct cmd_line *cmd)
 			perror("file open");
 			return -1;
 		}
+		info_msg("Input file \"%s\" opened.\n", cmd->input);
 	} else if (cmd->src_scheme == PATH_NET) {
 		/* open udp port for receive */
 		cmd->src_fd = udp_open(cmd);
@@ -391,6 +392,7 @@ open_files(struct cmd_line *cmd)
 
 			return -1;
 		}
+		info_msg("Output file \"%s\" opened.\n", cmd->output);
 	} else if (cmd->dst_scheme == PATH_NET) {
 		/* open udp port for send path */
 		cmd->dst_fd = udp_open(cmd);
diff --git a/test/mxc_vpu_test/vpu_test.h b/test/mxc_vpu_test/vpu_test.h
index cd6903c..2495787 100644
--- a/test/mxc_vpu_test/vpu_test.h
+++ b/test/mxc_vpu_test/vpu_test.h
@@ -41,7 +41,6 @@
 	} while (0)
 
 /*#define TVOUT_ENABLE*/
-/* Available on MX37. If not defined, then use VPU rotation.  */
 
 typedef unsigned long u32;
 typedef unsigned short u16;
-- 
1.8.0

